tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -0,0 +1,605 @@
+            });
+        });
+    }
+
+    /// <summary>
+    /// Removes all DevDriveOptimizedListViewModels from the list view model list and removes the dev drive
+    /// selected to apply the configuration to. This should refresh the UI to show no dev drives.
+    /// </summary>
+    private void RemoveDevDriveOptimizedListViewModels()
+    {
+        var totalLists = _devDriveOptimizedViewModelList.Count;
+        for (var i = totalLists - 1; i >= 0; i--)
+        {
+            _devDriveOptimizedViewModelList.RemoveAt(i);
+        }
+
+        ShouldShowOptimizedCollectionView = false;
+        DevDriveOptimizedCollectionView.Refresh();
+    }
+
+    /// <summary>
+    /// Starts the process of getting the list of DevDrives from all providers. the sync and next
+    /// buttons should be disabled when work is being done.
+    /// </summary>
+    private void GetDevDrives()
+    {
+        // We need to run this on a background thread so we don't block the UI thread.
+        Task.Run(async () =>
+        {
+            await _dispatcher.EnqueueAsync(async () =>
+            {
+                // Remove any existing DevDrivesListViewModels from the list if they exist. E.g when sync button is
+                // pressed.
+                RemoveDevDrivesListViewModels();
+
+                // Disable the sync and next buttons while we're getting the dev drives.
+                DevDriveLoadingCompleted = false;
+                UpdateNextButtonState();
+
+                // load the dev drives so we can show them in the UI.
+                await LoadAllDevDrivesInTheUI();
+
+                // Enable the sync and next buttons when we're done getting the dev drives.
+                UpdateNextButtonState();
+                DevDrivesCollectionView.Refresh();
+            });
+        });
+    }
+
+    /// <summary>
+    /// Removes all DevDrivesListViewModels from the list view model list and removes the dev drive
+    /// selected to apply the configuration to. This should refresh the UI to show no dev drives.
+    /// </summary>
+    private void RemoveDevDrivesListViewModels()
+    {
+        var totalLists = _devDriveViewModelList.Count;
+        for (var i = totalLists - 1; i >= 0; i--)
+        {
+            _devDriveViewModelList.RemoveAt(i);
+        }
+
+        // Reset the filter text and the selected provider name.
+        SelectedDevDriveProviderComboBoxName = _allKeyWordLocalized;
+        ShouldShowCollectionView = false;
+        DevDrivesCollectionView.Refresh();
+        ProviderComboBoxNamesCollectionView.Refresh();
+    }
+
+    /// <summary>
+    /// Adds a DevDrivesListViewModel from the DevDriveManager.
+    /// </summary>
+    private void AddListViewModelToList(DevDrivesListViewModel listViewModel)
+    {
+        // listViewModel doesn't exist so add it to the list.
+        _devDriveViewModelList.Add(listViewModel);
+        ShouldShowCollectionView = true;

```

Comment:
What does `ShouldShowCollectionView` control?  Show the view if the list is not empty?

Reply:
This controls the Visibility of DevDrivesListViewModel objects. While this is not useful in this PR, I have work planned in future to show the insights conditionally. So keeping this for now.